nexus/src/db/subnet_allocation.rs

```diff
@@ -250,15 +250,30 @@ fn push_null_if_overlapping_ip_range(
 /// SELECT *
 /// FROM candidate, candidate_ipv4, candidate_ipv6
 /// ```
-pub struct FilterConflictingVpcSubnetRangesQuery(pub VpcSubnet);
+pub struct FilterConflictingVpcSubnetRangesQuery {
+    subnet: VpcSubnet,
+    ipv4_block: ipnetwork::IpNetwork,
+    ipv6_block: ipnetwork::IpNetwork,
+}
+
+impl FilterConflictingVpcSubnetRangesQuery {
+    pub fn new(subnet: VpcSubnet) -> Self {
+        let ipv4_block = ipnetwork::IpNetwork::from(subnet.ipv4_block.0 .0);
+        let ipv6_block = ipnetwork::IpNetwork::from(subnet.ipv6_block.0 .0);
+        Self { subnet, ipv4_block, ipv6_block }
+    }
+}
```

Comment:
Is this to make the calling code inside `walk_ast` cleaner, or is it giving the parsed `ipv4_block` and `ipv6_block` longer lifetimes? Whichever it is, a comment would be helpful.

Reply:
Yeah, this is what I touched upon in the `Lifetimes` portion of this PR message . I left a similar comment below: 

https://github.com/oxidecomputer/omicron/blob/c6a43a7249a6de2283b85c3539aae65f8b5dea3b/nexus/src/db/subnet_allocation.rs#L1127-L1133

I can include that here too.